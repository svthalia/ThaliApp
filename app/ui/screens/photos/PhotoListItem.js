import React from 'react';
import PropTypes from 'prop-types';
import { Image, TouchableHighlight, ViewPropTypes } from 'react-native';
import SquareView from '../../components/memberView/SquareView';
import styles from './style/PhotoListItem';

const PhotoListItem = props => (
  <TouchableHighlight
    style={styles.touchable}
    onPress={() => props.onPress()}
  >
    <SquareView style={props.style} size={props.size} >
      <Image
        source={{
          uri: props.photo.file.small,
          headers: {
            Authorization: `Token ${props.token}`,
          },
        }}
        style={[
          styles.image,
          {
            width: props.size,
            height: props.size,
            transform: [{ rotate: `${props.photo.rotation}deg` }],
          },
        ]}
      />
    </SquareView>
  </TouchableHighlight>
);

PhotoListItem.propTypes = {
  size: PropTypes.number.isRequired,
  style: ViewPropTypes.style,
  photo: PropTypes.shape({
    pk: PropTypes.number.isRequired,
    rotation: PropTypes.number,
    hidden: PropTypes.bool,
    album: PropTypes.number,
    file: PropTypes.shape({
      full: PropTypes.string.isRequired,
      large: PropTypes.string.isRequired,
      medium: PropTypes.string.isRequired,
      small: PropTypes.string.isRequired,
    }).isRequired,
    size: PropTypes.string,
  }).isRequired,
  token: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

PhotoListItem.defaultProps = {
  style: {},
};

export default PhotoListItem;
