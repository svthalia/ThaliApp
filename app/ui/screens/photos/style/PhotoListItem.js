import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  touchable: {
  },
  image: {
  },
  overlayGradient: {
    position: 'absolute',
    top: '50%',
    bottom: 0,
    left: 0,
    right: 0,
    opacity: 0.6,
  },
});

export default styles;
