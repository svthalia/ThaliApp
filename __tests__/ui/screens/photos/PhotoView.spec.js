import React from 'react';
import renderer from 'react-test-renderer';
import PhotoListItem from '../../../../app/ui/screens/photos/PhotoListItem';

describe('PhotoView component', () => {
  const photo = {
    file: {
      small: 'http://test.local/small.png',
      medium: 'http://test.local/medium.png',
      large: 'http://test.local/large.png',
      full: 'http://test.local/full.png',
    },
    album: 0,
    pk: 0,
    size: 10,
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<PhotoListItem onPress={() => {}} photo={photo} size={20} token="token" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
