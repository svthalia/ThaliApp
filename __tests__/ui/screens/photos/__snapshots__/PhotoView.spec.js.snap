// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoView component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "height": 20,
          "width": 20,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "headers": Object {
            "Authorization": "Token token",
          },
          "uri": "http://test.local/small.png",
        }
      }
      style={
        Array [
          Object {},
          Object {
            "height": 20,
            "width": 20,
          },
        ]
      }
    />
  </View>
</View>
`;
